### PR TITLE
Update PR template to remind users to write tests

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Update PR template to remind users to write tests.

## How is this patch tested?

NA

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -9,9 +9,15 @@ Please reference any related feature requests, issues, or PRs here. For example,
 
 (Please fill in changes proposed in this fix)
 
-## How is this patch tested?
+## Test
```
Comment (harupy) [inline .github/pull_request_template.md:12]: Better section title?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -11,7 +11,15 @@ Please reference any related feature requests, issues, or PRs here. For example,
 
 ## How is this patch tested?
 
-(Details)
+<!--
+If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask. We'd be happy to help.
+-->
+- [ ] I've written tests for this PR using an MLflow-adopted testing framework (`pytest` for Python, `JUnit` for Java, `testthat` for R, or `jest` for JavaScript).
+- [ ] I've manually confirmed the proposed changes work properly.
```
Comment (BenWilson2) [inline .github/pull_request_template.md:18]: Could we make it a little bit more clear about when each of these approaches should be done?
I could see users selecting the 2nd box for new features
i.e., 

New Feature: 
[ ] I have added unit tests that ensure this feature works... (the wording you used is good) 
[ ] I have validated that unit tests pass locally for the test module that I have added to (e.g., I added a feature to the tracking server and made sure that the test suite that validates this behavior passes locally - some wording like that)

Bug-fix:
[ ] I have confirmed that existing tests pass locally and that the issue is resolved on a local MLflow environment (or something like that)

Comment (harupy) [reply .github/pull_request_template.md:18]: Thanks for the comment! I think tests are required in both cases.

Comment (harupy) [reply .github/pull_request_template.md:18]: We could do something like this as you suggested, but this makes the pull request template very long.

```
# new feature

- [ ] ...
- [ ] ...

# bug fix

- [ ] ...

# doc update

- [ ] ... 

```

Comment (harupy) [reply .github/pull_request_template.md]: > Should we add something about UI updates "I made sure that if I changed anything in the UI that it renders correctly"?

Does `confirmed the proposed feature/bug-fix/change works` include that?

**Review by BenWilson2 — APPROVED** [review]
LGTM! This reads well.

**Pull request merged**